hail/src/main/resources/include/hail/ArraySorter.h

```diff
@@ -0,0 +1,66 @@
+#ifndef HAIL_ARRAYSORTER_H
+#define HAIL_ARRAYSORTER_H 1
+
+#include <vector>
+#include <algorithm>
+
+namespace hail {
+
+template <typename ArrayBuilder, typename LessThan>
+class ArraySorter {
+  public:
+    using ArrayImpl = typename ArrayBuilder::ArrayImpl;
+    using T = typename ArrayImpl::T;
+
+    struct SortableElem {
+      T elem_;
+      LessThan lt_;
```

Comment:
I don't think we should be embedding a copy of the comparison functor into every element. If `LessThan` happened to have non-zero size, we're now inflating the entire collection. I think the following should work fine:
```c++
std::stable_sort(non_missing_.begin(), non_missing_.end(), LessThan())
```